### DISCUSSION TITLE
Star rejection

### DISCUSSION THREAD

**Discussion by AstroKeith** [smroid/cedar-detect#3] [General]
I've noticed that the brightest star(s) in an image can be rejected by cedar_detect.extract_centroids(). This is important for me to avoid.

Are they rejected purely on their absolute width ('blur') or relative to others in the image? Or is it the shape of the point spread function. I've tried reducing the exposure in steps, and it seems it might be the later effect? It seems that whenever a star intensity starts maxing out at 255 for a few pixels across, the star gets ignored? If so a simple auto exposure will fix it!

TIA, Keith

**Comment by smroid** [comment]
The cedar-detect algorithm requires that along a row, a center pixel (star candidate) at location C must be strictly brighter than the pixel at C-2 and C+2. Thus, if your bright star is over-exposed such that a horizontal cut has more than 3 saturated pixels, it won't detect. This sounds consistent with what you observed as you varied exposure.

The overall Cedar system does use auto-exposure to limit over-exposure of stars. In aim mode it adjusts exposure to achieve around 20-30 detected stars (adequate for robust plate solve), which usually results in a low likelihood of the brightest star being blown out. But if one or two bright stars are blown out by auto-exposure in order to get 20-30 stars overall, the plate solver will likely still work.

**Reply by smroid** [reply]
Another thing you can look at is to tell Cedar-detect to use binning, in which case long runs of saturated pixels become correspondingly shorter, and might thus satisfy the C-2..C+2 criterion.
